src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -407,16 +407,42 @@ public TimeSpan ReceiveDataTimeout
 
             set
             {
-                if (value != Timeout.InfiniteTimeSpan && (value <= TimeSpan.Zero || value > s_maxTimeout))
-                {
-                    throw new ArgumentOutOfRangeException(nameof(value));
-                }
-
+                CheckTimeSpanPropertyValue(value);
                 CheckDisposedOrStarted();
                 _receiveDataTimeout = value;
             }
         }
 
+        public bool TcpKeepAliveEnabled { get; set; }
```

Comment:
Shouldn't the set here call CheckDisposedOrStarted?

Reply:
The default here is false?  I'm just questioning that because the comment earlier says "The default settings when a TCP socket is initialized sets the keep-alive timeout to 2 hours and the keep-alive interval to 1 second" which suggests the default here should actually be true?

Reply:
> The default settings when a TCP socket is initialized sets the keep-alive timeout to 2 hours and the keep-alive interval to 1 second

[The same docs](https://docs.microsoft.com/en-us/windows/win32/winsock/so-keepalive#remarks) also state that :

> The SO_KEEPALIVE option for a socket is disabled (set to FALSE) by default. [...] When this socket option is enabled, the TCP stack sends keep-alive packets when no data or acknowledgement packets have been received for the connection within an interval. 

From this I concluded that if `SO_KEEPALIVE ` or `SIO_KEEPALIVE_VALS` are left unchanged, no keepalive packets gonna be sent. We may want to double check with the winsock team since the wording is kinda contradictional.

Reply:
Let's double-check, and then either update the implementation or the comment :)

Reply:
The defaults is false. This discussion resulted in an update of the docs ;)

MicrosoftDocs/win32#635

Reply:
I never see TCP keep-alive packets in HTTP network traces I get from customers, and that would be pretty obvious with the 1 second interval. I know we decided we're sticking with winsock defaults and violating RFC, but the RFC *does* say TCP keep-alives should default off in no uncertain terms:

```
            Implementors MAY include "keep-alives" in their TCP
            implementations, although this practice is not universally
            accepted.  If keep-alives are included, the application MUST
            be able to turn them on or off for each TCP connection, and
            they MUST default to off.
```

https://tools.ietf.org/html/rfc1122#page-101

I'm glad the winsock defaults at least align with the RFC on this part.